From 0547703fdaaf4691b98dc909ce38cd4cab89a0d2 Mon Sep 17 00:00:00 2001
From: neeraj solanki <neeraj.solanki@intel.com>
Date: Wed, 30 Mar 2022 11:00:55 +0530
Subject: [PATCH] Resolve the Youtube VP9 video streaming issue.

Issue: Youtube VP9 video playback got stuck with Android Cloud
stack (OneVPL-OMX) due to failure in frame decoding.

Modified the sync point for sync operation when device
is busy during frame decoding operation. It resolves the
Youtube VP9 video playback issue.

Change-Id: I73956107bfa5658230ebb42846dc0651fba31aa4
Tracked-On: OAM-101547
Signed-off-by: neeraj solanki <neeraj.solanki@intel.com>
---
 omx_components/src/mfx_omx_vdec_component.cpp | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/omx_components/src/mfx_omx_vdec_component.cpp b/omx_components/src/mfx_omx_vdec_component.cpp
index 7d86c326a3c8..4df0897e0fd2 100644
--- a/omx_components/src/mfx_omx_vdec_component.cpp
+++ b/omx_components/src/mfx_omx_vdec_component.cpp
@@ -3221,24 +3221,16 @@ mfxStatus MfxOmxVdecComponent::DecodeFrame(void)
                 MFX_OMX_LOG_INFO_IF(g_OmxLogLevel,"pOutSurface, CropW: %u, CropH: %u, Width: %u, Height: %u, TimeStamp: %llu",
                         pOutSurface->Info.CropW, pOutSurface->Info.CropH, pOutSurface->Info.Width, pOutSurface->Info.Height, pOutSurface->Data.TimeStamp);
             }
+
             if (MFX_WRN_DEVICE_BUSY == mfx_res)
             {
                 mfxStatus mfx_sts = MFX_ERR_NONE;
                 MFX_OMX_AUTO_TRACE("mfx(DecodeFrameAsync)::MFX_WRN_DEVICE_BUSY");
-                mfxSyncPoint* pDevBusySyncPoint = m_pSurfaces->GetSyncPoint();
-                if (pDevBusySyncPoint)
-                {
 #ifdef USE_ONEVPL
-                    mfx_sts = MFXVideoCORE_SyncOperation(m_Session, *pDevBusySyncPoint, MFX_OMX_INFINITE);
+                mfx_sts = MFXVideoCORE_SyncOperation(m_Session, *pSyncPoint, MFX_OMX_INFINITE);
 #else
-                    mfx_sts = m_Session.SyncOperation(*pDevBusySyncPoint, MFX_OMX_INFINITE);
+                mfx_sts = m_Session.SyncOperation(*pSyncPoint, MFX_OMX_INFINITE);
 #endif
-                }
-                else
-                {
-                    m_pDevBusyEvent->TimedWait(1);
-                    m_pDevBusyEvent->Reset();
-                }
                 if (MFX_ERR_NONE != mfx_sts)
                 {
                     MFX_OMX_LOG_ERROR("SyncOperation failed - %s", mfx_omx_code_to_string(mfx_sts));
-- 
2.17.1

